framework/src/araui/animation/ui_animation.c

```diff
@@ -511,7 +504,7 @@ static bool _ui_anim_scale_func(ui_widget_t widget, ui_anim_t anim, uint32_t *dt
 
 	intrp_scale = scale_anim_body->intrp_func(anim_body->t, scale_anim_body->from_scale, scale_anim_body->to_scale - scale_anim_body->from_scale, anim_body->d);
 
-	ui_widget_set_scale(widget, intrp_scale, intrp_scale);
+	ui_widget_set_scale_sync((ui_widget_body_t *)widget, intrp_scale, intrp_scale);
```

Comment:
In order to avoid svace issue, please check the return value.

Reply:
Fixed